mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -317,7 +318,12 @@ final class OneVsRest @Since("1.4.0") (
     val numClasses = MetadataUtils.getNumClasses(labelSchema).fold(computeNumClasses())(identity)
     instr.logNumClasses(numClasses)
 
-    val multiclassLabeled = dataset.select($(labelCol), $(featuresCol))
+    val multiclassLabeled = getClassifier match {
+      // SPARK-21306: cache weightCol if necessary
+      case c: HasWeightCol if c.isDefined(c.weightCol) && c.getWeightCol.nonEmpty =>
+        dataset.select($(labelCol), $(featuresCol), c.getWeightCol)
+      case _ => dataset.select($(labelCol), $(featuresCol))
+    }
```

Comment:
```OneVsRest``` is a classification estimator, I think we should make ```weightCol``` a member param of it like ```featuresCol```. For example:
```
val dataset = dataset       // This dataset has column: a, b, c.
val ova = new OneVsRest().setFeaturesCol("a").setClassifier(new LogisticRegression().setFeaturesCol("b"))
```
The features column used by ```OneVsRest``` is ```a```. The features column set for ```OneVsRest``` will override corresponding set in ```OneVsRest.classifier```. So we should follow this way for ```weightCol``` as well. Thanks.